stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -466,11 +466,19 @@ function get_type(sym::Expr, fn::Module)
     val, found = try_get_type(sym, fn)
     found && return val, found
     # https://github.com/JuliaLang/julia/issues/27184
-    if isexpr(sym, :macrocall)
+    newsym = if isexpr(sym, :macrocall)
         _, found = get_type(first(sym.args), fn)
         found || return Any, false
+        try
+            Meta.lower(fn, sym)
+        catch e
+            e isa LoadError && e.error isa MethodError && return Any, false
+            rethrow()
+        end
```

Comment:
```suggestion
        try
            Meta.lower(fn, sym)
        catch # macro expansion can throw an arbitrary error, just ignore it
            return Any, false
        end
```

Would be also nice if you add test cases that demonstrate it:
```julia
# @unsafemacro.[TAB]
macro unsafemacro()
    throw(ArgumentError("zero argument"))
end
# @unsafemacro("foo").[TAB]
macro unsafemacro(a)
    if isa(a, Number)
        return a
    end
    throw(ArgumentError(a))
end
```

Reply:
Thank you for the thorough review, I'll amend the PR accordingly! However, may I suggest to keep the `e isa LoadError` condition (barring the `&& e.error isa MethodError` part), since this `try`-`catch` block is only supposed to catch lowering errors, be they caused by invalid macro calls or macro calls expanding into errors.
I think any other error should leak because they are not supposed to happen, so they should be reported. But please let me know if I am misunderstanding something.

Reply:
You can see the fixed commit here: https://github.com/JuliaLang/julia/pull/43541/commits/6d9ae078efaa654b56692c02b67feaf284eefdbd

Reply:
My view is that REPL completion is just one application, and it should be implemented "safely". For example, Pluto and Juno use this module for its code completion feature, but they assume this module never throws.
So from this point of view, REPL completion ideally shouldn't involve any kind of actual code execution and potential side effect (including macroexansion, dictionary indexing, etc.). Nevertheless we're doing some of such "dangerous" operations in REPL completions since they can provide useful information, but potential errors from such inspections should be caught since it is not users but us (REPL completion implementers) who execute user code here.

Reply:
Ops, I think I missed your point. Yeah, it sounds reasonable to keep the `LoadError` condition.

Reply:
Yes, whatever code the user may write, I agree that REPL completion should definitely not crash on it. But if I understand correctly, `Meta.lower` always wraps whatever error may be caused by macro expansion in a `LoadError`, so if another kind of error is thrown, then something wrong is happening with julia itself, independently of the user's code. And as such, it would be best if it was reported, even if REPL completion may not be the best place to do so.
But I won't object further if you still feel like REPL completion is not the place for this, I can see your point as well so please let me know!

Reply:
Ah, it seems we crossed answers :)

Reply:
yep, sorry for the confusion :P
It would be appreciated if you could leave some comment here so that why we want to have this `rethrow` here.

Reply:
Comment added, see https://github.com/JuliaLang/julia/pull/43541/commits/af9f78abe94fbdd1b48fe20934dd484284b66e97